src/components/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext, useState, useRef,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { TodosContext } from '../contexts/TodosContext';
+import { TodoStatus } from '../types/TodoStatus';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const editInputRef = useRef<HTMLInputElement>(null);
+
+  const { id, title, completed } = todo;
+
+  const handleTodoDelete = (todoId: number) => {
+    const filteredTodos
+      = todos.filter(currentTodo => currentTodo.id !== todoId);
```

Comment:
```suggestion
    const filteredTodos = todos
      .filter(currentTodo => currentTodo.id !== todoId);
```
Don't linebreak the assignment